### PR TITLE
Minor doc changes to the consensus algorithms

### PR DESCRIPTION
**Pull request by bangerth**

### CODE REVIEW

File: include/deal.II/base/mpi_consensus_algorithms.h [hunk 1]
```diff
@@ -166,15 +169,22 @@ namespace Utilities
       class Interface
       {
       public:
+          /**
+           * Constructor. `process` is an object that provides information
+           * about what processes the current process wants to communicate with,
+           * and the data to be sent/received. `comm` is the communicator on
+           * which this communication is to happen.
+           */
```
Comment (marcfehling) [inline include/deal.II/base/mpi_consensus_algorithms.h:177]: `indent` complains
```suggestion
        /**
         * Constructor. `process` is an object that provides information
         * about what processes the current process wants to communicate with,
         * and the data to be sent/received. `comm` is the communicator on
         * which this communication is to happen.
         */
```

**Review by peterrum — APPROVED** [review]

**Comment by bangerth** [comment]
I added another commit that adds a lengthy documentation of what "consensus algorithms" actually are and are trying to accomplish. I had to understand that from the code and thought I'd write up my understanding of it.

File: include/deal.II/base/mpi_consensus_algorithms.h [hunk 2]
```diff
@@ -141,23 +144,69 @@ namespace Utilities
...
+       * be sent, and to come up with an algorithm that ensures that this
+       * communication happens.
+       *
+       * That this is not a trivial problem can be seen by an analogy of the
+       * postal service. There, some senders may request information from some
+       * other participants in the postal service. So they send a letter that
+       * requests the information, but the recipients do not know how many such
+       * letters they need to expect (or that they should expect any at all).
+       * They also do not know how long they need to keep checking their mailbox
+       * for incoming requests. They can be considered reliable, however: We
```
Comment (peterrum) [inline include/deal.II/base/mpi_consensus_algorithms.h:162]:
```suggestion
       * for incoming requests, however, can be considered reliable: We
```
Currently, it sounds like that the requests are reliable.

File: include/deal.II/base/mpi_consensus_algorithms.h [hunk 3]
```diff
@@ -166,15 +215,22 @@ namespace Utilities
       class Interface
       {
       public:
+        /**
+         * Constructor. `process` is an object that provides information
```
Comment (peterrum) [inline include/deal.II/base/mpi_consensus_algorithms.h:219]:
```suggestion
         * Constructor. @p process is an object that provides information
```

**Review by peterrum — APPROVED** [review]
Thanks a lot!

**Comment by bangerth** [comment]
Thanks, @peterrum . I addressed your comments.

**Review by peterrum — APPROVED** [review]

**Pull request merged**